src/anchoredinterval.jl

```diff
@@ -166,21 +165,11 @@ end
 Base.convert(::Type{T}, interval::AnchoredInterval{P, T}) where {P, T} = anchor(interval)
 
 # Date/DateTime attempt to convert to Int64 instead of falling back to convert(T, ...)
-Base.Date(interval::AnchoredInterval{P, Date}) where P = convert(Date, interval)
-Base.DateTime(interval::AnchoredInterval{P, DateTime}) where P = convert(DateTime, interval)
+Compat.Dates.Date(interval::AnchoredInterval{P, Date}) where P = convert(Date, interval)
+Compat.Dates.DateTime(interval::AnchoredInterval{P, DateTime}) where P = convert(DateTime, interval)
 
 ##### DISPLAY #####
 
-Base.show(io::IO, ::Type{HourEnding}) = print(io, "HourEnding{T}")
-Base.show(io::IO, ::Type{HourBeginning}) = print(io, "HourBeginning{T}")
-
-Base.show(io::IO, ::Type{HourEnding{T}}) where T <: TimeType = print(io, "HourEnding{$T}")
-Base.show(io::IO, ::Type{HourBeginning{T}}) where T <: TimeType = print(io, "HourBeginning{$T}")
-
-function Base.show(io::IO, ::Type{AnchoredInterval{P, T}}) where {P, T}
-    print(io, "AnchoredInterval{$P, $T}")
-end
```

Comment:
I think the intention was to avoid these being fully qualified on Julia 0.6. Should we just have a version check around this to limit it's usage to Julia 0.6?

Reply:
The problem is that it's always a Bad Thing To Do<sup>TM</sup>, it's just penalized with actual crashing on 0.7.

Reply:
Why is it always a bad thing to do? I'm mostly just curious so I can give some rational against doing this.

Reply:
See https://github.com/invenia/Intervals.jl/issues/31, in particular the link to Jeff's comment. Basically it's redefining what it means to call `show` on a `Type`.

Reply:
Thanks